Put GRMHD equations to the system struct docs

yoonso0-0 opened the pull request:
## Proposed changes

Add a full description of GRMHD equations that we are evolving to the documentation of the `system` struct. I tried to adopt the notations and variables from our group's recent paper (https://arxiv.org/abs/2109.12033). Also added conversion factors for magnetic fields to `Hydro/Units.hpp`.

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

yoonso0-0 committed:
Add geometrized units for magnetic fields

yoonso0-0 committed:
Add docs to GRMHD system struct

ermost commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp:
```diff
@@ -35,6 +35,138 @@ namespace grmhd {
+ * \sqrt{1+\gamma^{ij}W^2v_iv_j} \f}
+ * - The specific enthalpy \f$h = 1 + \epsilon + p/\rho\f$ where \f$p\f$ is the
+ * pressure specified by a particular equation of state (EoS)
+ * - The comoving magnetic field
```
Maybe say `(in the ideal MHD limit)`, since this is not the the general comoving field expression.
You can also add the general definition, 
`\f$b^b = -^*F^{ba}u_a \f$`

nilsdeppe commented on src/PointwiseFunctions/Hydro/Units.hpp:
```diff
@@ -36,6 +36,11 @@ constexpr double rest_mass_density_unit =
 /// `pressure_cgs = pressure_geometric * pressure_unit`
 constexpr double pressure_unit =
     mass_unit / (length_unit * time_unit * time_unit);
+/// CGS Gaussian unit of magnetic field.
```
Please add something like `pressure_cgs = pressure_geometric * pressure_unit` to the documentation so it's clear what should be done.

nilsdeppe commented on src/Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp:
```diff
@@ -35,6 +35,138 @@ namespace grmhd {
+ * \f{align*}
+ *  \sqrt{4\pi}\frac{c^4}{G^{3/2}M_\odot} \approx 8.35 \times 10^{19}
+ *  \,\text{Gauss}
+ * \f}
```
Maybe add `That is, sqrt{b^2}_Gauss = sqrt{b^2}_code` or something like that to be explicit about how to convert?

nilsdeppe reviewed (changes requested):
Thanks for doing this! Please squash things directly

yoonso0-0 commented:
The fixup commit is only for the `system` struct documentation.

I changed the first commit (a new entry to Hydro/Units.hpp) into a new one, since I thought the previous one was still prone to confusion. I think we can keep only a single value of unit solely accounting for the dimension (length, mass, time) of "Gauss" without including the factor of `sqrt(4pi)`. I added an example on that.

nilsdeppe reviewed (commented):
LGTM! Thanks for doing this!

ermost commented:
Thanks for including that comment, Yoonsoo!

yoonso0-0 commented:
Thanks for the review! :)
